src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,31 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+
+/**
+ * Represent a deadline event
+ */
+public class Deadline extends Task {
+
+    private final LocalDate deadline;
+
+    Deadline(String desc, String deadline) {
+        super(desc);
+        this.deadline = LocalDate.parse(deadline);
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + deadline.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+    }
+
+    /**
+     * @return the representation of the event when written to disk
+     */
+    @Override
```

Comment:
`toDisk` gives the impression that this method is writing the task to the disk. Maybe `toDiskFormat` would be a better way to put it?